website_and_docs/content/documentation/grid/advanced_features/customize_node.en.md

```diff
@@ -0,0 +1,206 @@
+---
+title: "Customizing a Node"
+linkTitle: "Customize Node"
+weight: 4
+aliases: [
+"/documentation/en/grid/grid_4/customize_node/",
+"/documentation/grid/advanced_features/customize_node/"
+]
+---
+
+
+### How to customize a Node
+
+There are times when we would like a node to be customized to our needs. 
+
+For e.g., we may like to do some additional setup before a test case begins execution and some clean-up after a test case runs to completion.
+
+Following steps can be followed for this:
+
+* Create a class that extends `org.openqa.selenium.grid.node.Node`
+* Add a static method (this will be our factory method) to the newly created class whose signature looks like this: 
+
+  `public static Node create(Config config)`. Here:
+
+    * `Node` is of type `org.openqa.selenium.grid.node.Node`
+    * `Config` is of type `org.openqa.selenium.grid.config.Config`
+* Within this factory method, include logic for creating your new Class.
+* To wire in this new customized logic into the hub, start the node and pass in the fully qualified class name of the above class to the argument `--node-implementation`
+
+Let's see an example of all this:
+
+1. Create a sample project using your favourite build tool (**Maven**|**Gradle**).
+2. Add the below two dependencies to your sample project.
+  * [org.seleniumhq.selenium/selenium-java](https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java)
+  * [org.seleniumhq.selenium/selenium-grid](https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-grid/4.6.0)
+3. Add your customized node to the project.
+4. Build an [uber jar](https://imagej.net/develop/uber-jars) if you would like to just use `java -jar` to start the node.
+5. Now start the node using the command:
+
+```bash
+java -jar selenium-server-4.6.0.jar node \
+--node-implementation org.seleniumhq.samples.CommentatingNode
```

Comment:
You also need to add the uber jar to the class path, which can be done with the `--ext` flag. Like we do it in docker selenium https://github.com/SeleniumHQ/docker-selenium/blob/trunk/Hub/start-selenium-grid-hub.sh#L32-L34

Reply:
No. This is not needed. The jar name is what confused you. I will fix it. Essentially, if one builds the customised node into an uber jar, then there's no need to add `--ext`. The `--ext` is only used if one builds their implementation separately and then adds it along with the existing selenium-server jar (like how we added the tracing required jars and the `selenium-http-jdk-client.jar` instead of bundling everything into the same uber/fat jar.

Reply:
You are right. Apologies.